subsys/net/lib/nrf_cloud/src/nrf_cloud_codec_internal.c

```diff
@@ -2509,7 +2512,7 @@ int nrf_cloud_gnss_msg_json_encode(const struct nrf_cloud_gnss_data * const gnss
 	    json_add_str_cs(gnss_msg_obj,
 			    NRF_CLOUD_JSON_MSG_TYPE_KEY,
 			    NRF_CLOUD_JSON_MSG_TYPE_VAL_DATA) ||
-	    ((gnss->ts_ms > NRF_CLOUD_NO_TIMESTAMP) &&
+	    ((gnss->ts_ms != NRF_CLOUD_NO_TIMESTAMP) &&
```

Comment:
align this check (and the one in `nrf_cloud_sensor_data_encode`) with the header comments for the various occurrences of `ts_ms`, which currently state:
```
If a negative value is provided, the timestamp will be ignored.
```

my preference would be to just make the timestamp checks:
```
if (ts_ms >= 0) { 
/* add timestamp */
}
```
and leave the comments as they are.

Reply:
You prefer that I not use `NRF_CLOUD_NO_TIMESTAMP`?

Reply:
the comment for the various `ts_ms` variables is:
`If a negative value is provided, the timestamp will be ignored.`
the code should match the comment.  i don't have a preference how it is done.
currently a value of -2 would result in a timestamp being added.

Reply:
OK, understood.